Allow different content types in openapi-response-validator for OpenAPI 3.0 response definitions

# The problem

Response validation fails for content types other than 'application/json'.

Affected package: openapi-response-validator getSchemas function

# Example

Following example spec fails for the GET /doc/admin or /doc/dev call when using the express-openapi middleware.

Error:
```json
{
  "errorCode": "type.openapi.responseValidation",
  "message": "response should be null"
}
```

## The Open API Spec

```json
{
  "openapi": "3.0.2",
  "info": {
    "version": "2.0.0",
    "title": "Basic Usage Example"
  },
  "paths": {}
}
```

## Path Handler

```javascript
/*******************************************************************
 *   paths/doc/{type}.js
 *******************************************************************/
const utilities = require('./../../../../../../../../../index');

module.exports = function (config, parameters, responses) {
     let operations = {
          GET
     };

     operations.GET.apiDoc = {
          summary: 'Returns documentation.',
          operationId: 'getDoc',
          parameters: [
               parameters.language,
               {
                    name: "type",
                    description: "type",
                    schema: {
                         type: "string",
                         enum: [
                              "admin",
                              "dev"
                         ]
                    },
                    in: "path",
                    required: true
               }
          ],
          responses: {
               206: {
                    description: "Forwarding markdown",
                    content: {
                         "text/markdown": {
                              schema: {
                                   type: "string"
                              }
                         }
                    }
               },
               default: responses.Error
          }
     };

     function GET(req, res, next) {
          let lang = req.headers["accept-language"] === 'en' ? 'en' : 'de';
          let type = req.params.type;
          let result = utilities.fs.loadFile("resources/documentation/" + lang + "/" + type + ".md").toString();
          let httpStatusCode = 206;
          let contentType = "text/markdown";
          res.validateAndSend(httpStatusCode, result, {"content-type": contentType});
     }

     return operations;
};
```

# Affected Open API Response validator function

The getSchemas function only works for content type 'application/json' and not e.g. for 'text/markdown'.

```javascript
/*******************************************************************
 *   /node_modules/openapi-response-validator/dist/index.js
 *******************************************************************/
function getSchemas(responses, definitions, components) {
    var schemas = {};
    Object.keys(responses).forEach(function (name) {
        var response = responses[name];
        var schema = response
            ? typeof response.schema === 'object'
                ? response.schema
                : typeof response.content === 'object' &&
                    typeof response.content['application/json'] === 'object' &&
                    typeof response.content['application/json'].schema === 'object'
                    ? response.content['application/json'].schema
                    : { type: 'null' }
            : { type: 'null' };
        schemas[name] = {
            $schema: 'http://json-schema.org/schema#',
            type: 'object',
            properties: {
                response: schema
            },
            definitions: definitions || {},
            components: components || {}
        };
    });
    return schemas;
}
```

My suggestion would be to always get the first key of the response.content object and not hardcoded 'application/json':

```javascript
typeof response.content[Object.keys(response.content)[0]] === 'object'
```

Struggling with the tooling of the openapi repository on Windows 10 using Babun with zsh 5.5.1 (i686-pc-cygwin).

When I  run ./bin/test I get following error:
```console
...
D:\development\workspaces\open-api\node_modules\.bin\_mocha.CMD:1
(function (exports, require, module, __filename, __dirname) { @IF EXIST "%~dp0\node.exe" (
                                                              ^

SyntaxError: Invalid or unexpected token
    at new Script (vm.js:74:7)
    at createScript (vm.js:246:10)
...
```

When I run ./bin/test packages/openapi-response-validator I get following error:
```console
> mocha --watch-extensions ts -w

No test files found
npm ERR! code ELIFECYCLE
...
```

Any ideas how to solve this?

--- a/packages/openapi-response-validator/index.ts
+++ b/packages/openapi-response-validator/index.ts
@@ -1,5 +1,6 @@
 import * as Ajv from 'ajv';
 import { IJsonSchema, OpenAPIV2, OpenAPIV3 } from 'openapi-types';
+
 const LOCAL_DEFINITION_REGEX = /^#\/([^\/]+)\/([^\/]+)$/;
 
 export interface IOpenAPIResponseValidator {
@@ -26,6 +27,7 @@ export interface OpenAPIResponseValidatorArgs {
       schema: OpenAPIV2.Schema | OpenAPIV3.SchemaObject;
     };
   };
+
   errorTransformer?(
     openAPIResponseValidatorValidationError: OpenAPIResponseValidatorError,
     ajvError: Ajv.ErrorObject
@@ -151,9 +153,11 @@ function getSchemas(responses, definitions, components) {
       ? typeof response.schema === 'object'
         ? response.schema
         : typeof response.content === 'object' &&
-          typeof response.content['application/json'] === 'object' &&
-          typeof response.content['application/json'].schema === 'object'
-        ? response.content['application/json'].schema
+          typeof response.content[Object.keys(response.content)[0]] ===
+            'object' &&
+          typeof response.content[Object.keys(response.content)[0]].schema ===
+            'object'
+        ? response.content[Object.keys(response.content)[0]].schema
         : { type: 'null' }
       : { type: 'null' };
 

--- a/packages/openapi-response-validator/test/data-driven/accept-openapi3-responses-different-content-types.js
+++ b/packages/openapi-response-validator/test/data-driven/accept-openapi3-responses-different-content-types.js
@@ -1,24 +1,23 @@
 module.exports = {
-    constructorArgs: {
-      responses: {
-        200: {
-          description: 'Ok',
-          content: {
-            'text/markdown': {
-              schema: {
-                type: 'string'
-              }
+  constructorArgs: {
+    responses: {
+      200: {
+        description: 'Ok',
+        content: {
+          'text/markdown': {
+            schema: {
+              type: 'string'
             }
           }
         }
-      },
-      definitions: null
+      }
     },
-  
-    inputStatusCode: 200,
-  
-    inputResponseBody: '# documentation\n',
-  
-    expectedValidationError: void 0
-  };
-  
+    definitions: null
+  },
+
+  inputStatusCode: 200,
+
+  inputResponseBody: '# documentation\n',
+
+  expectedValidationError: void 0
+};

--- a/packages/openapi-response-validator/test/data-driven/accept-openapi3-responses-different-content-types.js
+++ b/packages/openapi-response-validator/test/data-driven/accept-openapi3-responses-different-content-types.js
@@ -1,0 +1,24 @@
+module.exports = {
+    constructorArgs: {
+      responses: {
+        200: {
+          description: 'Ok',
+          content: {
+            'text/markdown': {
+              schema: {
+                type: 'string'
+              }
+            }
+          }
+        }
+      },
+      definitions: null
+    },
+  
+    inputStatusCode: 200,
+  
+    inputResponseBody: '# documentation\n',
+  
+    expectedValidationError: void 0
+  };
+  

